cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -50,10 +50,8 @@ type ScaleSet struct {
 	minSize int
 	maxSize int
 
-	sizeMutex         sync.Mutex
-	curSize           int64
-	lastSizeRefresh   time.Time
-	sizeRefreshPeriod time.Duration
```

Comment:
The `sizeRefreshPeriod` was always 0 after #3717. The TTL for VMSS cache refreshes is in honoured there where we depend on the CA core `cloudprovider.Refresh()` calls to update the caches.

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/azure/azure_manager.go#L166-L168